processor/factory.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processor
+
+import "github.com/spf13/viper"
+
+// TraceDataProcessorFactory is an interface that builds a new TraceDataProcessor based on
+// some viper.Viper configuration.
+type TraceDataProcessorFactory interface {
```

Comment:
I would prefer to have this with a different name. Names that I have in my mind "component", "middleware", "stage", "processingstage"

Reply:
You mean renaming `TraceDataProcessor` -> `TraceProcessingStage`?

Reply:
I am fine with the current name, I realized that I was not clear in my thoughts.